File: init_param.R
fill.matrix<-function(expr,nrow=1,ncol=1){
  matrix(eval(expr,envir=list(x=nrow*ncol)),nrow=nrow,ncol=ncol)
}
init_param<-function(layer_dim){
  L<-length(layer_dim)
  parameters=vector("list",0)
  for (l in 1:(L-1)){
    n=layer_dim[l]*layer_dim[l+1]
    parameters[[paste0("W",l)]]<-fill.matrix(rnorm(n=n)*0.01,nrow=layer_dim[l+1],ncol=layer_dim[l])
    parameters[[paste0("b",l)]]<-matrix(0,layer_dim[l+1],1)
  }
  return(parameters)
}

File: init_param.jl
function init_param(layer_dim)
  L = length(layer_dim)
  parameters = Dict()
  for l in 2:(L)
    parameters[string("W", l-1)] = rand(layer_dim[l], layer_dim[l - 1]) * 0.01
    parameters[string("b", l-1)] = zeros(layer_dim[l], 1)
    @assert size(parameters[string("W", l-1)]) == (layer_dim[l], layer_dim[l - 1]) "Weights size wrong in init_param function"
    @assert size(parameters[string("b", l-1)]) == (layer_dim[l], 1) "Bias size wrong in init_param function"
  end
  return parameters
end